mapquester_backend/pois/views.py

```diff
@@ -1,32 +1,106 @@
+from django.contrib.auth import get_user_model
+from django.core.files.base import ContentFile
 from django.shortcuts import render, redirect, get_object_or_404
 from django.contrib import messages
 from .models import POI, PoiManager
 from django.contrib.auth.decorators import login_required
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework import status
+# import base64
+# import json
 
 # Create POI View
-@login_required
+
+User = get_user_model()
+
+
+#@login_required
+@api_view(['POST'])
 def create_poi(request):
-    if request.method == 'POST':
-        name = request.POST.get('name')
-        description = request.POST.get('description')
-        category = request.POST.get('category')
-        coordinates = request.POST.get('coordinates')
-        attachments = request.FILES.get('attachments')
-
-        # Create POI instance
-        poi = POI(
-            name=name,
-            description=description,
-            category=category,
-            coordinates=coordinates,
-            attachments=attachments
-        )
-        poi.save()
+    # Extract and validate JSON payload
+    data = request.data
+    user_id = data.get("userId")
+    name = data.get("name")
+    description = data.get("description")
+    category = data.get("category")
+    coordinates = data.get("coordinates")
+    attachments = data.get("attachments")
```

Comment:
You should use serializers for getting and validating the payloads. The serializer can be setup as separate class

https://www.django-rest-framework.org/api-guide/serializers/